CONTRIBUTING.md

```diff
@@ -76,6 +76,15 @@ src/CheckBasicStyle.lua 1>&2 -g
 
 Note that the check script is not smart enough to catch everything, so not having any warnings does not necessarily imply that you followed the conventions fully. The other humans working on this will perform more checks before merging.
 
+Github-related conventions
+--------------------------
+The following rules are not strictly enforced, but if you follow them, you will make it easier for us to review your changes.
+ - Use good short description for your PRs. `Store Health as float` is good, `Fix #4024` is not good.
```

Comment:
I agree with having descriptive commit titles, but is there any reason to discourage adding issue references to commit descriptions?

Reply:
For me it seems that naming a commit "fixes #something" rather than "store health as float" only makes sense when viewed on github, and not anywhere else. When you just take the git repo, you should have a description about what the commit is doing not just a reference to any bug tracker. Imo.

Reply:
I absolutely agree that all commits should have descriptive titles, but if there is a relevant issue, I think it really ought to be mentioned. Just to be clear, here is my idea of an 'ideal' commit message:

    Store entity health as float rather than short
    
    Originally changed in vanilla 1.x. 
    Fixes #4024

Reply:
I concur, the title should describe it on its own, but the description should include "fixes #id" -- among other things, the "fixes" magic word actually has [special behavior](https://help.github.com/articles/closing-issues-using-keywords/) (the issue is automatically closed when merged) that is useful to preserve.

Reply:
I agree, but most people don't read commit descriptions (so you don't really notice it but github does? (or does github? Idk)) and even less people use commit descriptions....